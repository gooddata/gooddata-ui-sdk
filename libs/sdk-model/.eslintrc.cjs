// (C) 2020 GoodData Corporation
module.exports = {
    parser: "@typescript-eslint/parser",
    plugins: ["prettier", "sonarjs", "eslint-plugin-tsdoc", "regexp"],
    extends: [
        "@gooddata",
        "plugin:import-esm/recommended",
        "plugin:sonarjs/recommended",
        "plugin:regexp/recommended",
        "../../.eslintrc.js",
    ],
    parserOptions: { tsconfigRootDir: __dirname },
};
