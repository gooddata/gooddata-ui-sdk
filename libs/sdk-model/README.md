# GoodData.UI domain models

[![npm version](https://img.shields.io/npm/v/@gooddata/sdk-model)](https://www.npmjs.com/@gooddata/sdk-model)&nbsp;
[![npm monthly downloads](https://img.shields.io/npm/dm/@gooddata/sdk-model)](https://npmcharts.com/compare/@gooddata/sdk-model?minimal=true)&nbsp;
![typescript](https://img.shields.io/badge/typescript-first-blue?logo=typescript)

This package is a part of the [GoodData.UI SDK](https://sdk.gooddata.com/gooddata-ui/docs/about_gooddataui.html).
To learn more, check [the source monorepo](https://github.com/gooddata/gooddata-ui-sdk).

This package provides domain models for GoodData.UI.

These domain models are backend-agnostic. This makes them reusable across different [Analytical Backend](https://www.npmjs.com/package/@gooddata/sdk-backend-spi) implementations.
The package includes TypeScript type definitions, factory functions, functions to get or set certain properties
of the objects in an immutable way, and more. These are used in both the `@gooddata/sdk-backend-*` and `@gooddata/sdk-ui* packages`.

## License

(C) 2017-2022 GoodData Corporation

This project is under MIT License. See [LICENSE](https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-model/LICENSE).
