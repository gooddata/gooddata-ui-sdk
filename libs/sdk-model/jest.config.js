// (C) 2019 GoodData Corporation
import base from "../../common/config/jest/jest.config.base.mjs";

export default {
  ...base,
};