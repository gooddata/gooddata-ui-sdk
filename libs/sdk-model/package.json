{
    "name": "@gooddata/sdk-model",
    "version": "8.12.3-beta.0",
    "author": "GoodData",
    "description": "GoodData Model definitions used by UI components and Backend SPI and its implementations",
    "repository": {
        "type": "git",
        "url": "https://github.com/gooddata/gooddata-ui-sdk.git",
        "directory": "libs/sdk-model"
    },
    "license": "MIT",
    "main": "dist/index.js",
    "module": "esm/index.js",
    "es2015": "esm/index.js",
    "browser": "dist/index.js",
    "typings": "esm/index.d.ts",
    "sideEffects": false,
    "files": [
        "dist/**/*.js",
        "dist/**/*.json",
        "dist/**/*.d.ts",
        "dist/**/*.map",
        "esm/**/*.js",
        "esm/**/*.json",
        "esm/**/*.d.ts",
        "esm/**/*.map"
    ],
    "config": {
        "eslint": "-c .eslintrc.js --ext ts src/"
    },
    "scripts": {
        "clean": "rm -rf ci dist esm coverage *.log && jest --clearCache",
        "dev": "tsc -p tsconfig.dev.json --watch",
        "build": "concurrently \"npm run build-cjs\" \"npm run build-esm\" && npm run api-extractor",
        "build-cjs": "tsc -p tsconfig.build.json",
        "build-esm": "tsc -p tsconfig.build.esm.json",
        "api-extractor": "mkdir -p api && [ -z \"${CI}\" ] && (api-extractor run -l) || (api-extractor run)",
        "test": "jest --watch",
        "test-once": "jest --maxWorkers=${JEST_MAX_WORKERS:-'45%'}",
        "test-ci": "JEST_JUNIT_OUTPUT=./ci/results/test-results.xml jest --ci --config jest.ci.js",
        "eslint": "eslint $npm_package_config_eslint",
        "eslint-ci": "mkdir -p ./ci/results && eslint -f checkstyle -o ci/results/eslint-results.xml $npm_package_config_eslint",
        "prettier-check": "prettier --check '{src,test}/**/*.{ts,tsx,json,scss,md,yaml,html}'",
        "prettier-write": "prettier --write '{src,test}/**/*.{ts,tsx,json,scss,md,yaml,html}'",
        "dep-cruiser": "depcruise --validate .dependency-cruiser.js --output-type err-long src/",
        "dep-cruiser-ci": "depcruise --validate .dependency-cruiser.js --output-type err-long src/",
        "validate": "npm run dep-cruiser && npm run eslint && npm run prettier-check",
        "validate-ci": "npm run dep-cruiser-ci && npm run eslint-ci && npm run prettier-check"
    },
    "dependencies": {
        "json-stable-stringify": "^1.0.1",
        "lodash": "^4.17.19",
        "spark-md5": "^3.0.0",
        "stringify-object": "^3.3.0",
        "ts-invariant": "^0.7.3",
        "tslib": "^2.0.0"
    },
    "devDependencies": {
        "@gooddata/eslint-config": "^4.1.0",
        "@microsoft/api-documenter": "^7.17.0",
        "@microsoft/api-extractor": "^7.20.0",
        "@types/jest": "^27.0.1",
        "@types/json-stable-stringify": "^1.0.32",
        "@types/lodash": "^4.14.158",
        "@types/spark-md5": "^3.0.1",
        "@types/stringify-object": "^3.2.0",
        "@typescript-eslint/eslint-plugin": "^5.45.0",
        "@typescript-eslint/parser": "^5.45.0",
        "concurrently": "^6.0.2",
        "dependency-cruiser": "^12.12.2",
        "eslint": "^8.25.0",
        "eslint-plugin-header": "^3.0.0",
        "eslint-plugin-import": "^2.22.0",
        "eslint-plugin-jest": "^27.1.1",
        "eslint-plugin-no-only-tests": "^2.4.0",
        "eslint-plugin-prettier": "^4.0.0",
        "eslint-plugin-regexp": "^1.9.0",
        "eslint-plugin-sonarjs": "^0.16.0",
        "eslint-plugin-tsdoc": "^0.2.14",
        "jest": "^27.5.1",
        "jest-junit": "^3.0.0",
        "prettier": "~2.5.0",
        "ts-jest": "^27.0.5",
        "typescript": "4.0.2"
    }
}
