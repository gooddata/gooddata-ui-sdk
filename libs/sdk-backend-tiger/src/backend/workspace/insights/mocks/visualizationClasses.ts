// (C) 2019-2024 GoodData Corporation
import { IVisualizationClass } from "@gooddata/sdk-model";
import sortBy from "lodash/sortBy.js";

export const visualizationClasses: IVisualizationClass[] = sortBy(
    [
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:scatter",
                orderIndex: 7,
                iconSelected: "local:scatter.selected",
                url: "local:scatter",
                title: "Scatter plot",
                identifier: "gdc.visualization.scatter",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/1",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:donut",
                orderIndex: 10,
                iconSelected: "local:donut.selected",
                url: "local:donut",
                title: "Donut chart",
                identifier: "gdc.visualization.donut",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/2",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:area",
                orderIndex: 4,
                iconSelected: "local:area.selected",
                url: "local:area",
                title: "Stacked Area chart",
                identifier: "gdc.visualization.area",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/3",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:table",
                orderIndex: 0,
                iconSelected: "local:table.selected",
                url: "local:table",
                title: "Table",
                identifier: "gdc.visualization.table",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/4",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:headline",
                orderIndex: 6,
                iconSelected: "local:headline.selected",
                url: "local:headline",
                title: "Headline",
                identifier: "gdc.visualization.headline",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/5",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:column",
                orderIndex: 1,
                iconSelected: "local:column.selected",
                url: "local:column",
                title: "Column chart",
                identifier: "gdc.visualization.column",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/6",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:line",
                orderIndex: 3,
                iconSelected: "local:line.selected",
                url: "local:line",
                title: "Line chart",
                identifier: "gdc.visualization.line",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/7",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:treemap",
                orderIndex: 12,
                iconSelected: "local:treemap.selected",
                url: "local:treemap",
                title: "Treemap",
                identifier: "gdc.visualization.treemap",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/8",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:pyramid",
                orderIndex: 12.5,
                iconSelected: "local:pyramid.selected",
                url: "local:pyramid",
                title: "Pyramid chart",
                identifier: "gdc.visualization.pyramid",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/1002",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:funnel",
                orderIndex: 12.6,
                iconSelected: "local:funnel.selected",
                url: "local:funnel",
                title: "Funnel chart",
                identifier: "gdc.visualization.funnel",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/1003",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:heatmap",
                orderIndex: 13,
                iconSelected: "local:heatmap.selected",
                url: "local:heatmap",
                title: "Heatmap",
                identifier: "gdc.visualization.heatmap",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/9",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:bubble",
                orderIndex: 8,
                iconSelected: "local:bubble.selected",
                url: "local:bubble",
                title: "Bubble chart",
                identifier: "gdc.visualization.bubble",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/10",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:pie",
                orderIndex: 9,
                iconSelected: "local:pie.selected",
                url: "local:pie",
                title: "Pie chart",
                identifier: "gdc.visualization.pie",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/11",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:bar",
                orderIndex: 2,
                iconSelected: "local:bar.selected",
                url: "local:bar",
                title: "Bar chart",
                identifier: "gdc.visualization.bar",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/12",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:combo",
                orderIndex: 5,
                iconSelected: "local:combo.selected",
                url: "local:combo2",
                title: "Combo",
                identifier: "gdc.visualization.combo",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/669",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:bullet",
                orderIndex: 19,
                iconSelected: "local:bullet.selected",
                url: "local:bullet",
                title: "Bullet",
                identifier: "gdc.visualization.bullet",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/1000",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:waterfall",
                orderIndex: 19.1,
                iconSelected: "local:waterfall.selected",
                url: "local:waterfall",
                title: "Waterfall chart",
                identifier: "gdc.visualization.waterfall",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/1119",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:dependencywheel",
                orderIndex: 19.2,
                iconSelected: "local:dependencywheel.selected",
                url: "local:dependencywheel",
                title: "Dependency wheel chart",
                identifier: "gdc.visualization.dependencywheel",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glrn/obj/1219",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:sankey",
                orderIndex: 19.3,
                iconSelected: "local:sankey.selected",
                url: "local:sankey",
                title: "Sankey chart",
                identifier: "gdc.visualization.sankey",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu5gpyn/obj/1319",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:pushpin",
                orderIndex: 20,
                iconSelected: "local:pushpin.selected",
                url: "local:pushpin",
                title: "Geo Pushpin",
                identifier: "gdc.visualization.pushpin",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/1001",
            },
        },
        {
            visualizationClass: {
                checksum: "local",
                icon: "local:repeater",
                orderIndex: 0.5,
                iconSelected: "local:repeater.selected",
                url: "local:repeater",
                title: "Repeater",
                identifier: "gdc.visualization.repeater",
                uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/1234",
            },
        },
    ],
    (cls) => cls.visualizationClass.orderIndex,
);
