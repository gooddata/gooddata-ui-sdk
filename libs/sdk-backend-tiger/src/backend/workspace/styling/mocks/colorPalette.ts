// (C) 2019-2020 GoodData Corporation
import { IColorPalette } from "@gooddata/sdk-model";

export const DefaultColorPalette: IColorPalette = [
    {
        guid: "1",
        fill: { r: 20, g: 178, b: 226 },
    },
    {
        guid: "2",
        fill: { r: 0, g: 193, b: 141 },
    },
    {
        guid: "3",
        fill: { r: 229, g: 77, b: 66 },
    },
    {
        guid: "4",
        fill: { r: 241, g: 134, b: 0 },
    },
    {
        guid: "5",
        fill: { r: 171, g: 85, b: 163 },
    },
    {
        guid: "6",
        fill: { r: 244, g: 213, b: 33 },
    },
    {
        guid: "7",
        fill: { r: 148, g: 161, b: 174 },
    },
    {
        guid: "8",
        fill: { r: 107, g: 191, b: 216 },
    },
    {
        guid: "9",
        fill: { r: 181, g: 136, b: 177 },
    },
    {
        guid: "10",
        fill: { r: 238, g: 135, b: 128 },
    },
    {
        guid: "11",
        fill: { r: 241, g: 171, b: 84 },
    },
    {
        guid: "12",
        fill: { r: 133, g: 209, b: 188 },
    },
    {
        guid: "13",
        fill: { r: 41, g: 117, b: 170 },
    },
    {
        guid: "14",
        fill: { r: 4, g: 140, b: 103 },
    },
    {
        guid: "15",
        fill: { r: 181, g: 60, b: 51 },
    },
    {
        guid: "16",
        fill: { r: 163, g: 101, b: 46 },
    },
    {
        guid: "17",
        fill: { r: 140, g: 57, b: 132 },
    },
    {
        guid: "18",
        fill: { r: 136, g: 219, b: 244 },
    },
    {
        guid: "19",
        fill: { r: 189, g: 234, b: 222 },
    },
    {
        guid: "20",
        fill: { r: 239, g: 197, b: 194 },
    },
];
