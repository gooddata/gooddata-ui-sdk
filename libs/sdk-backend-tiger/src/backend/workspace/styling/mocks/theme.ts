// (C) 2019-2022 GoodData Corporation
import { ITheme } from "@gooddata/sdk-model";

export const DefaultTheme: ITheme = {};
