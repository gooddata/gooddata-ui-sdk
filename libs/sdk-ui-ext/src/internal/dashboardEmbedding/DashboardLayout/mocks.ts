// (C) 2007-2021 GoodData Corporation
import { idRef } from "@gooddata/sdk-model";
import { IFluidLayoutSizeByScreen, IWidget } from "@gooddata/sdk-backend-spi";
import { IDashboardViewLayout, IDashboardViewLayoutColumn } from "./interfaces/dashboardLayout";
import { DashboardViewLayoutWidgetClass } from "./interfaces/dashboardLayoutSizing";

export type DashboardLayoutRowMock<TCustomContent> = {
    title?: string;
    description?: string;
    columns: Array<[TCustomContent, IFluidLayoutSizeByScreen?]>;
};

export const dashboardWidgetMock = (
    id: string,
    widgetClass: DashboardViewLayoutWidgetClass = "bar",
): IWidget => {
    const commonProps: Omit<IWidget, "kpi" | "insight" | "type"> = {
        description: "",
        drills: [],
        ignoreDashboardFilters: [],
        identifier: id,
        ref: {
            identifier: id,
        },
        title: `${widgetClass} ${id}`,
        uri: id,
    };

    return widgetClass === "kpi"
        ? {
              ...commonProps,
              type: "kpi",
              kpi: {
                  comparisonType: "none",
                  metric: idRef("measure"),
              },
          }
        : {
              ...commonProps,
              insight: idRef("insight"),
              type: "insight",
          };
};

export const dashboardRowMock = <TCustomContent>(
    columns: Array<[TCustomContent, IFluidLayoutSizeByScreen?]>,
    title?: string,
    description?: string,
): DashboardLayoutRowMock<TCustomContent> => {
    return {
        columns,
        title,
        description,
    };
};

export const dashboardLayoutMock = <TCustomContent>(
    rowMocks: DashboardLayoutRowMock<TCustomContent>[],
): IDashboardViewLayout<TCustomContent> => {
    const emptyLayout: IDashboardViewLayout<TCustomContent> = {
        type: "fluidLayout",
        rows: [],
    };

    return rowMocks.reduce((acc: IDashboardViewLayout<TCustomContent>, rowMock, rowIndex) => {
        if (!acc.rows[rowIndex]) {
            acc.rows[rowIndex] = {
                columns: [],
                ...(rowMock.title || rowMock.description
                    ? {
                          header: {
                              title: rowMock.title,
                              description: rowMock.description,
                          },
                      }
                    : {}),
            };
        }

        return rowMock.columns.reduce(
            (
                acc2: IDashboardViewLayout<TCustomContent>,
                [content, size = { xl: { widthAsGridColumnsCount: 12 } }],
                columnIndex,
            ) => {
                const column: IDashboardViewLayoutColumn<TCustomContent> = {
                    size,
                    content,
                };
                acc2.rows[rowIndex].columns[columnIndex] = column;
                return acc2;
            },
            acc,
        );
    }, emptyLayout);
};

export const allWidgetClasses: DashboardViewLayoutWidgetClass[] = [
    "alluvial",
    "area",
    "bar",
    "bubble",
    "bullet",
    "column",
    "combo",
    "combo2",
    "donut",
    "funnel",
    "geo",
    "headline",
    "heatmap",
    "histogram",
    "line",
    "pareto",
    "pushpin",
    "pie",
    "scatter",
    "table",
    "treemap",
    "waterfall",
    "xirr",
    "kpi",
];
