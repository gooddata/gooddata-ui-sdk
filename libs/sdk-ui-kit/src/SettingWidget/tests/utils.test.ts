// (C) 2022 GoodData Corporation

import { ITheme } from "@gooddata/sdk-model";
import { getColorsPreviewFromTheme } from "../StylingSettingWidget/index.js";
import { customItemsMock, fullyDefinedThemeMock } from "./mocks.js";
import { describe, it, expect } from "vitest";

describe("getColorsPreviewFromTheme", () => {
    it("should return default array of colors when theme is empty", () => {
        const result = getColorsPreviewFromTheme({});

        expect(result).toMatchSnapshot();
    });

    it("should return correct array of colors when only complementary colors are set in theme", () => {
        const result = getColorsPreviewFromTheme(customItemsMock[0].content as ITheme);

        expect(result).toMatchSnapshot();
    });

    it("should return correct array of colors when theme is fully defined", () => {
        const result = getColorsPreviewFromTheme(fullyDefinedThemeMock);

        expect(result).toMatchSnapshot();
    });
});
