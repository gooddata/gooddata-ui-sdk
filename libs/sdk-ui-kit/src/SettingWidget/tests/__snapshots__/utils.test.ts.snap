// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getColorsPreviewFromTheme > should return correct array of colors when only complementary colors are set in theme 1`] = `
[
  "#323232",
  "#292727",
  "#14b2e2",
  "#EAEAEA",
  "#D6D6D6",
  "#B9B9B9",
]
`;

exports[`getColorsPreviewFromTheme > should return correct array of colors when theme is fully defined 1`] = `
[
  "#f0f0f0",
  "#dfdfdf",
  "#eba12a",
  "#fff",
  "#ff0000",
  "#778491",
]
`;

exports[`getColorsPreviewFromTheme > should return default array of colors when theme is empty 1`] = `
[
  "#303442",
  "#fff",
  "#14b2e2",
  "#000",
  "#464e56",
  "#778491",
]
`;
